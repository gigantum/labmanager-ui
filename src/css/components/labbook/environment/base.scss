.Base *{
  font-weight: 100;

}

.Base__header-container{
  @include flex(left, row)
}

.Base__header{
  color: $romanSilver;
}

.Base__card{
  color: $romanSilver;
  @extend .card;
  @include flex(space-between, row);
  img{
    min-width: 70px;
    margin: 0 10px 0 0;
  }

  padding: 30px;
}


.Base__image-container{
  width: 250px;
  margin: 0 20px 0 0;
  @include flex(space-between, row);

  p{
    font-size: 14px;
    font-weight: 400;
  }

  .Base__name{
    font-weight: 400;
  }

  .Base__title{
    flex: 1;
    width: 160px;
  }
}

.Base__card-text{
  flex: 1;

  p{
    margin-top: 0px;
  }
}

.Base__categories{
  @include flex(flex-start, row);

  h6{
    font-size: 16px;
    font-weight: 400;
  }
}

.Base__categories-languages{
  width: 50%;
}

.Base__categories-tools{
  width: 50%;
}
