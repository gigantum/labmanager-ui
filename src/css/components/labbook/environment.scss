.Environment{
  overflow: auto;
  height: calc(100% - 13px);
  width: 100%;
}

.Environment__header-container{
  @include flex(left, row)
}

.Environment__header{
  font-weight: light;
  color: $font-grey;
  font-size: 22px;
}

.Environment__edit-button,
.Environment__edit-button:hover,
.Environment__edit-button:focus{
  font-weight: light;
  color: $font-grey;
  font-size: 14px;
  height: 30px;
  width: 30px;
  padding: 5px;
  border: 0px;
  background: url('./../images/icons/edit.svg') no-repeat center center;
  background-size: 22px;
}

.Environment__card{
  background-color: $white;
  padding: 10px;
  color: $charcoal;
  width: 330px;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 1px 5px 0 rgba(0, 0, 0, 0.12), 0 3px 1px -2px rgba(0, 0, 0, 0.2);
}


.Environment__image-container{
  align-items: center;
  align-content: center;
  align-self: center;
}

.Environment__card-text{
  //background-color: $grey-light;
  padding: 10px;
  width: 200px;
}

.Environment__edit-container{
  align-self: center;
  margin-left: 20px;
}

.Environment__modal{
  position: fixed;
  left: 50%;
  top: 50%;
  z-index: 100;
  width: 800px;
  height: 600px;

  margin: -300px 0 0 -400px;

  border: 1px solid $grey;
  background-color: $grey-light;
  box-shadow: 10px 10px 5px #888888;

  .SelectBaseImage__progress-button{
    display: none;
  }

}

.Environment__modal-close{
  position: absolute;
  top: 5px;
  right: 5px;
  cursor: pointer;
}

.Environment__package-dependencies{
  margin: 10px 20px 10px 0;

  .Environment__card-text{
      background-color: $white;
      width: 150px;
      box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 1px 5px 0 rgba(0, 0, 0, 0.12), 0 3px 1px -2px rgba(0, 0, 0, 0.2);
      p{
        margin: 0 5px 0 0;
      }

  }
}

.Environment__development-environment-item,
.Environment__dependencies{
  margin: 0 20px 0 0;
}
