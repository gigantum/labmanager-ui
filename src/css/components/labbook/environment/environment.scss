.Environment {
  width: 100%;
}

.Environment__header-container {
  @include flex(left, row)
}

.Environment__header {
  font-weight: light;
  color: $font-grey;
  font-size: 22px;
}

.Environment__edit-button,
.Environment__edit-button:hover,
.Environment__edit-button:focus {
  font-weight: light;
  color: $font-grey;
  font-size: 14px;
  height: 30px;
  width: 30px;
  padding: 5px;
  border: 0px;
  background: url('./../images/icons/edit.svg') no-repeat center center;
  background-size: 22px;
}

.Environment__card {
  background-color: $white;
  padding: 10px;
  color: $jet;
  width: 100%;
  @extend .box-shadow;
}


.Environment__image-container {
  align-items: center;
  align-content: center;
  align-self: center;
}

.Environment__card-text {
  //background-color: $grey-light;
  padding: 10px;
  width: 200px;
}

.Environment__edit-container {
  align-self: center;
  margin-left: 20px;
}

.Environment__modal {
  position: fixed;
  @include modal(500px, 800px);

  .SelectDevelopmentEnvironment__title,
  .AddEnvironmentPackage__title {
    font-size: 1.5em;
    text-align: center;
    border-bottom: 1px solid $azure;
    padding: 5px;
    color: $azure;
  }

  .SelectBaseImage__progress-button {
    display: none;
  }

}

.Environment__modal-close {
  cursor: pointer;
  @include close();
}

.Environment__package-dependencies {
  margin: 10px 20px 10px 0;

  .Environment__card-text{
      background-color: $white;
      width: 150px;
      @extend .box-shadow;

      p{
        margin: 0 5px 0 0;
      }

  }
}

.Environment__development-environment-item,
.Environment__dependencies {
  margin: 0 20px 0 0;
}
