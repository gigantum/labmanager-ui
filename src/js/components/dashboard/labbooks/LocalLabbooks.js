//vendor
import React, { Component } from 'react'
import {
  createPaginationContainer,
  graphql
} from 'react-relay'
//components
import WizardModal from 'Components/wizard/WizardModal'
import Loader from 'Components/shared/Loader'
import LocalLabbookPanel from 'Components/dashboard/labbooks/LocalLabbookPanel'
import ImportModule from 'Components/import/ImportModule'




let isLoadingMore = false;

class LocalLabbooks extends Component {

  constructor(props){
  	super(props);

    this._goToLabbook = this._goToLabbook.bind(this)
    this._loadMore = this._loadMore.bind(this)

  }

  /**
  * fires when a componet mounts
  * adds a scoll listener to trigger pagination
  */
  componentDidMount() {
    let that = this;
    window.addEventListener('scroll', function(e){
      let root = document.getElementById('root')
      let distanceY = window.innerHeight + document.documentElement.scrollTop + 200,
          expandOn = root.offsetHeight;

      if(that.props.feed.localLabbooks){
        if ((distanceY > expandOn) && !isLoadingMore && that.props.feed.localLabbooks.pageInfo.hasNextPage) {
            that._loadMore(e);
        }
      }
    });
  }
  /**
  *  @param {string} labbookName - inputs a labbook name
  *  routes to that labbook
  */
  _goToLabbook = (labbookName) => {
    this.setState({'labbookName': labbookName})

    this.props.history.replace(`/labbooks/${labbookName}`)
  }

  /**
  *  @param {event} e
  *  loads more labbooks using the relay pagination container
  */
  _loadMore = (e) => {
    isLoadingMore = true
    if(e){
      e.preventDefault();
    }
    this.props.relay.loadMore(
      10, // Fetch the next 10 feed items
      (ev) => {
        isLoadingMore = false;
      }
    );
  }

  render(){
      let {props} = this;

      if(props.feed.localLabbooks){

        return(
          <div className="LocalLabbooks">

            <WizardModal
              ref="wizardModal"
              handler={this.handler}
              history={this.props.history}
              {...props}
            />

            <div className="LocalLabbooks__title-bar flex flex--row justify--space-between">
              <h4 className="LocalLabbooks__title" onClick={()=> this.refs.wizardModal._showModal()} >
                Lab Books
                <div className="LocalLabbooks__title-add"></div>
              </h4>

            </div>
            <div className='LocalLabbooks__labbooks'>
              <div className="LocalLabbooks__sizer">
              <div
                key={'addLabbook'}
                onClick={()=> this.refs.wizardModal._showModal()}
                className="LocalLabbooks__panel LocalLabbooks__panel--add flex flex--row justify--center">
                <div
                  // onClick={()=> this._openImport()}
                  className="LocalLabbooks__labbook-icon">
                    <div className="LocalLabbooks__title-add"></div>
                </div>
              </div>

              <ImportModule
                  ref="ImportModule_localLabooks"
                  {...props}
                  className="LocalLabbooks__panel LocalLabbooks__panel--import" />

              {

                this.props.feed.localLabbooks.edges.map((edge) => {

                  return (
                    <LocalLabbookPanel
                      key={edge.node.name}
                      ref={'LocalLabbookPanel' + edge.node.name}
                      className="LocalLabbooks__panel"
                      edge={edge}
                      goToLabbook={this._goToLabbook}/>
                  )
                })
              }
            </div>

          </div>

        </div>
      )
    }else{
      return(<Loader />)
    }

  }
}

export default createPaginationContainer(
  LocalLabbooks,
  {feed: graphql`
      fragment LocalLabbooks_feed on Query{
        localLabbooks(first: $first, after:$cursor)@connection(key: "LocalLabbooks_localLabbooks"){
          edges {
            node {
              name
              description
              owner{
                id
                username
              }
              environment{
                id
                imageStatus
                containerStatus
              }
            }
            cursor
          }
          pageInfo {
            endCursor
            hasNextPage
            hasPreviousPage
            startCursor
          }
        }
      }
    `,
  },
  {
    direction: 'forward',
    getConnectionFromProps(props) {
      return props.feed.localLabbooks
    },
    getFragmentVariables(prevVars, first, cursor) {
      return {
        ...prevVars,
        first: first
      };
    },
    getVariables(props, {first, cursor}, fragmentVariables) {
      first = 10;
      cursor = props.feed.localLabbooks.pageInfo.endCursor;
      return {
        first,
        cursor
        // in most cases, for variables other than connection filters like
        // `first`, `after`, etc. you may want to use the previous values.
      };
    },
    query: graphql`
      query LocalLabbooksPaginationQuery(
        $first: Int!
        $cursor: String
      ) {
          ...LocalLabbooks_feed
      }
    `
  }
);
