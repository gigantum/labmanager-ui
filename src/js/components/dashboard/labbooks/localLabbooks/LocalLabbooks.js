//vendor
import React, { Component } from 'react'
import {
  createPaginationContainer,
  graphql
} from 'react-relay'
//components
import LocalLabbookPanel from 'Components/dashboard/labbooks/localLabbooks/LocalLabbookPanel'
import LabbooksPaginationLoader from '../labbookLoaders/LabbookPaginationLoader'
import ImportModule from 'Components/import/ImportModule'
//helpers
import ContainerLookup from './ContainerLookup'


export class LocalLabbooks extends Component {
  constructor(props){
    super(props)
    this.state = {
      sort: this.props.sort,
      reverse: this.props.reverse,
      isPaginating: false,
      refetchLoading: false,
      containerList: null,
    }

    this._captureScroll = this._captureScroll.bind(this)
    this._loadMore = this._loadMore.bind(this)
    this._refetch = this._refetch.bind(this);
    this._containerLookup = this._containerLookup.bind(this)
  }

  componentWillReceiveProps(nextProps) {
    if(nextProps.sort !== this.state.sort || nextProps.reverse !== this.state.reverse) {
      this.setState({sort: nextProps.sort, reverse: nextProps.reverse});
      this._refetch(nextProps.sort, nextProps.reverse);
    }
  }

  componentDidMount() {
    if(!this.props.loading){
      if(this.props.wasSorted) {
      this._refetch(this.state.sort, this.state.reverse);
      }
      this.props.sortProcessed()
      window.addEventListener('scroll', this._captureScroll);
      this._containerLookup();
    }
  }

  _containerLookup(){
    let self = this;
    let idArr = this.props.localLabbooks.localLabbooks.edges.map(edges =>edges.node.id)
    ContainerLookup.query(idArr).then((res)=>{
      if(res && res.data && res.data.labbookList && res.data.labbookList.localById){
        self.setState({containerList: res.data.labbookList.localById})
        this.containerLookup = setTimeout(()=>{
          self._containerLookup()
        }, 10000)
      }
    })
  }


  componentWillUnmount() {
    clearTimeout(this.containerLookup)
    window.removeEventListener("scroll", this._captureScroll)
  }

  /**
    *  @param {}
    *  fires when user scrolls
    *  if nextPage exists and user is scrolled down, it will cause loadmore to fire
  */
  _captureScroll = () => {
    let root = document.getElementById('root')
    let distanceY = window.innerHeight + document.documentElement.scrollTop + 200,
        expandOn = root.offsetHeight;
    if(this.props.localLabbooks.localLabbooks){
      if ((distanceY > expandOn) && !this.state.isPaginating && this.props.localLabbooks.localLabbooks.pageInfo.hasNextPage) {
        this._loadMore();
      }
    }
  }

  /**
    * @param {string, boolean} sort reverse
    * fires when parent _refetch function is called
    * causes relay to refetch with new parameters
  */
  _refetch(sort, reverse){
    let self = this;
    let relay = self.props.relay;
    this.setState({refetchLoading: true})
    this.props.changeRefetchState(true)

    relay.refetchConnection(
      20,
      (res, err)=>{
        if(err){
          console.log(err)
        }
        this.setState({refetchLoading: false})
        this.props.changeRefetchState(false)

      },
      {first: 100,
        cursor: null,
        sort: sort,
        reverse: reverse,
      }
    )
  }

  /**
    *  @param {}
    *  loads more labbooks using the relay pagination container
  */

  _loadMore = () => {

    this.setState({
      'isPaginating': true
    })
    if(this.props.localLabbooks.localLabbooks.pageInfo.hasNextPage){
      this.props.relay.loadMore(
        10, // Fetch the next 10 items
        (ev) => {
          this.setState({
            'isPaginating': false
          })
        }
      );
    }
  }

  render(){
    if((this.props.localLabbooks && this.props.localLabbooks.localLabbooks && this.props.localLabbooks.localLabbooks.edges) || this.props.loading){

      let labbooks = !this.props.loading ? this.props.filterLabbooks(this.props.localLabbooks.localLabbooks.edges, this.props.filterState) : [];

      return(
        <div className='LocalLabbooks__labbooks'>
        <div className="LocalLabbooks__sizer grid">

          <ImportModule
              ref="ImportModule_localLabooks"
              {...this.props}
              showModal={this.props.showModal}
              className="LocalLabbooks__panel column-4-span-3 LocalLabbooks__panel--import"
          />
          {

            !this.state.refetchLoading && labbooks.map((edge, index) => {
              return (
                <LocalLabbookPanel
                  key={edge.node.name}
                  ref={'LocalLabbookPanel' + edge.node.name}
                  className="LocalLabbooks__panel"
                  edge={edge}
                  history={this.props.history}
                  environment={this.state.containerList && this.state.containerList[index].environment}
                  goToLabbook={this.props.goToLabbook}/>
              )
            })
          }
          {
            Array(5).fill(1).map((value, index) => {

                return (
                  <LabbooksPaginationLoader
                    key={'LocalLabbooks_paginationLoader' + index}
                    index={index}
                    isLoadingMore={this.state.isPaginating || this.props.loading ||this.state.refetchLoading}
                  />
              )
            })
          }
        </div>
      </div>
      )

    }else{
      return(<div></div>)
    }
  }
}

export default createPaginationContainer(
  LocalLabbooks,
  graphql`
    fragment LocalLabbooks_localLabbooks on LabbookList{
      localLabbooks(first: $first, after: $cursor, sort: $sort, reverse: $reverse)@connection(key: "LocalLabbooks_localLabbooks", filters: []){
        edges {
          node {
            id
            name
            description
            owner
            creationDateUtc
          }
          cursor
        }
        pageInfo {
          endCursor
          hasNextPage
          hasPreviousPage
          startCursor
        }
      }
    }
  `,
  {
    direction: 'forward',
    getConnectionFromProps(props, error) {
      return props.localLabbooks.localLabbooks
    },
    getFragmentVariables(prevVars, first, cursor) {
      return {
        ...prevVars,
        first: first
      };
    },
    getVariables(props, {first, cursor, sort, reverse}, fragmentVariables) {
      first = 10;
      cursor = props.localLabbooks.localLabbooks.pageInfo.endCursor;
      sort = fragmentVariables.sort;
      reverse = fragmentVariables.reverse
      return {
        first,
        cursor,
        sort,
        reverse
        // in most cases, for variables other than connection filters like
        // `first`, `after`, etc. you may want to use the previous values.
      };
    },
    query: graphql`
      query LocalLabbooksPaginationQuery(
        $first: Int!
        $cursor: String
        $sort: String
        $reverse: Boolean
      ) {
        labbookList{
          ...LocalLabbooks_localLabbooks
        }
      }
    `
  }
);
