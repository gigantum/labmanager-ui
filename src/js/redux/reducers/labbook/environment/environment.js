/**
 * constants
 */
export const CLOSE_ENVIRONMENT_MENUS = 'CLOSE_ENVIRONMENT_MENUS'
export const TOGGLE_PACKAGE_MENU = 'TOGGLE_PACKAGE_MENU'
export const TOGGLE_CUSTOM_MENU = 'TOGGLE_CUSTOM_MENU'
export const RESET_DETAIL_STORE = 'RESET_DETAIL_STORE'
export const CONTAINER_MENU_WARNING = 'CONTAINER_MENU_WARNING'
export const UPDATE_CONTAINER_MENU_VISIBILITY = 'UPDATE_CONTAINER_MENU_VISIBILITY'



export default (
 state = {
   'status': "",
   'containerMenuOpen': false,
   'containerMenuWarning': ''
 },
 action
) => {

if (action.type === CLOSE_ENVIRONMENT_MENUS) {
   return {
     ...state,
     packageMenuVisible: false,
     viewContainerVisible: false
   };
 }else if (action.type === TOGGLE_PACKAGE_MENU) {

   return {
     ...state,
     packageMenuVisible: action.payload.packageMenuVisible
   };
 }else if (action.type === TOGGLE_CUSTOM_MENU) {
   return {
     ...state,
     viewContainerVisible: action.payload.viewContainerVisible
   };
 }else if(action.type === RESET_DETAIL_STORE){
   return {
     ...state,
     detailMode: false
   };
 } else if(action.type === CONTAINER_MENU_WARNING) {
  return {
    ...state,
    containerMenuWarning: action.payload.message
  }
} else if (action.type === UPDATE_CONTAINER_MENU_VISIBILITY) {
  
  return {
    ...state,
    containerMenuOpen: action.payload.containerMenuOpen
  };
}

 return state;
};
