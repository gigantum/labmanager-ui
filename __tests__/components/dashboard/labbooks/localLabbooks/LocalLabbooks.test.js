import React from 'react'
import renderer from 'react-test-renderer';
import {configure, shallow, mount} from 'enzyme';
import sinon from 'sinon';
import history from 'JS/history'
import {StaticRouter, Link} from 'react-router';
import json from './__relaydata__/LocalLabbooks.json'
import LocalLabbooks from 'Components/dashboard/labbooks/localLabbooks/LocalLabbooks';
import relayTestingUtils from 'relay-testing-utils'
import {MemoryRouter } from 'react-router-dom'
import environment from 'JS/createRelayEnvironment'
import {Provider} from 'react-redux'
import store from 'JS/redux/store'
import {BrowserRouter as Router} from 'react-router-dom'

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const variables = {first:20}

store.dispatch({
  type: 'SET_FILTER_TEXT',
  payload: {
    filterText: ''
  }
})

const showModal = () =>{

}

const goToLabbook = () =>{

}

const filterLabbooks = (labbooks, filter) =>{
  let filteredLabbooks = [];
  let username = localStorage.getItem('username')
  if(filter === username){
    filteredLabbooks = labbooks.filter((labbook)=>{
        return (labbook.node.owner === username)
    })

  }else if(filter === "others"){
    filteredLabbooks = labbooks.filter((labbook)=>{
        return (labbook.node.owner !== username)
    })
  }else{
    filteredLabbooks = labbooks;
  }

  return filteredLabbooks
}

const changeRefetchState = () =>{

}

const sortProcessed = () => {

}

const loadMore = (props, value, ha) => {

  let labbooks = json.data.labbookList.localLabbooks
  labbooks.edges = labbooks.edges.slice(0, 5)
  return labbooks
}

let labbookList = json.data.labbookList
labbookList.localLabbooks.edges = labbookList.localLabbooks.edges.slice(0, 5)

const fixtures = {
  localLabbooks: labbookList,
  wasSorted: false,
  sort: 'modified_on',
  reverse: false,
  labbookListId: json.data.labbookList.id,
  filterState: 'all',
  section: 'local',
  loading: false,
  showModal,
  goToLabbook,
  filterLabbooks,
  changeRefetchState,
  sortProcessed,
  history
}


describe('LocalLabbooks', () => {

  it('LocalLabbooks snapshot', () => {

    const localLabbooksSnap = renderer.create(

       relayTestingUtils.relayWrap(
        <Provider store={store}>
          <Router>
            <LocalLabbooks

              {...fixtures}

            />
          </Router>
        </Provider>, {}, json.data.labbookList)

    )

    const tree = localLabbooksSnap.toJSON()

    expect(tree).toMatchSnapshot()

  })

  /****
   *
   * Shallow
   * Run shallow tests here
   *
   *
   *****/
  const localLabbooksShallow = mount(
    <Provider store={store}>
     <Router>
      <LocalLabbooks history={history} {...fixtures} feed={json.data}/>
     </Router>
    </Provider>

  );


  it('LocalLabbooks panel length', () => {


    expect(localLabbooksShallow.find('.LocalLabbooks__panel')).toHaveLength(16)
  })



  /****
   *
   * Mount
   * Run shallow tests here
   *
   *
   *****/
  const showModalTest = sinon.spy();
  const goToLabbookTest = sinon.spy();
  const sortProcessedTest = sinon.spy();

  const localLabbooksMount = mount(
    relayTestingUtils.relayWrap(
      <Provider store={store}>
        <Router>
          <LocalLabbooks
            {...fixtures}
            showModal={showModalTest}
            goToLabbook={goToLabbookTest}
            sortProcessed={sortProcessedTest}
            relay={{loadMore: loadMore}}
          />
       </Router>
      </Provider>, {}, json.data.labbookList
     )
  );

  it('Simulates sort processed', () => {

    expect(sortProcessedTest).toHaveProperty('callCount', 0);
  });

  it('Simulates opening create labbook', () => {

    localLabbooksMount.find('.Labbooks__labbook-button').simulate('click');

    expect(showModalTest).toHaveProperty('callCount', 1);
  });


  it('Simulates opening a labbook', () => {

    localLabbooksMount.find('.LocalLabbooks__panel').at(4).simulate('click');


    expect(goToLabbookTest).toHaveProperty('callCount', 1);
  });


  // it('Simulates pagination', () => {
  //
  //   // console.log(localLabbooksMount)
  //   // console.log( localLabbooksMount.instance())
  //   //window.dispatchEvent(new window.UIEvent('scroll', { detail: 1800}))
  //   //localLabbooksMount.instance()._loadMore()
  //
  //   // expect(window.screenTop).toBe(1800)
  // });



  // describe('LocalLabbooks load more', () => {
  //   const relay = {loadMore: () => {}}
  //
  //   expect(localLabbooksMount.find('.LocalLabbooks__panel')).toHaveLength(22)
  // })
  //
  // describe('LocalLabbooks show modal', () => {
  //   const relay = {loadMore: () => {}}
  //   const localLabbooks = mount(
  //
  //      <LocalLabbooks relay={relay} history={history} {...fixtures} feed={json.data}/>
  //   );
  //
  //   localLabbooks.find('.LocalLabbooks__title').simulate('click')
  //
  //   expect(localLabbooks.node.refs.wizardModal.state.modal_visible).toBeTruthy()
  //
  // })
  //
  // describe('LocalLabbooks show modal by panel', () => {
  //   const relay = {loadMore: () => {}}
  //   const localLabbooks = mount(
  //
  //      <LocalLabbooks relay={relay} history={history} {...fixtures} feed={json.data}/>
  //   );
  //   localLabbooks.find('.LocalLabbooks__panel--add').simulate('click')
  //
  //   expect(localLabbooks.node.refs.wizardModal.state.modal_visible).toBeTruthy()
  //
  // })
  //
  //
  //
  // describe('Test LocalLabbooks click', () => {
  //
  //   const localLabbooks = mount(
  //
  //       <LocalLabbooks history={history} feed={json.data}/>
  //
  //   );
  //
  //   localLabbooks.at(0).simulate('click')
  //   localLabbooks.setState({'labbookName': json.data.localLabbooks.edges[0].node.name})
  //
  //   expect(localLabbooks.node.state.labbookName === json.data.localLabbooks.edges[0].node.name).toBeTruthy();
  // })
  //
  //
  // describe('Test LocalLabbooks edges output', () => {
  //
  //   const localLabbooks = mount(
  //
  //       <LocalLabbooks history={history} feed={json.data}/>
  //
  //   );
  //
  //   expect(localLabbooks.find('.LocalLabbooks__labbooks .LocalLabbooks__text-row h4').at(0).text()).toEqual(json.data.localLabbooks.edges[0].node.name)
  //
  // })
  // describe('Test scroll functon', () => {
  //
  //   const localLabbooks = mount(
  //
  //       <LocalLabbooks history={history} feed={json.data}/>
  //
  //   );
  //
  //   localLabbooks.find('.LocalLabbooks__panel').at(20).simulate('scroll')
  //
  //
  //
  //   expect(localLabbooks.find('.LocalLabbooks__panel')).toHaveLength(22)
  //
  // })
})



export default variables
