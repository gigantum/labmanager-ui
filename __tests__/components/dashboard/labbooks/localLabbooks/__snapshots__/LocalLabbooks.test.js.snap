// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalLabbooks LocalLabbooks snapshot 1`] = `
<div
  className="Labbooks__listing"
>
  <div
    className="grid"
  >
    <div
      className="ImportModule Card Card--add Card--import column-4-span-3"
      id="dropZone"
      onDragEnd={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      type="file"
    >
      <div
        className="Import__labbook-main"
      >
        <div
          className="Import__labbook-header"
        >
          <div
            className="Import__labbook-icon"
          >
            <div
              className="Import__add-icon"
            />
          </div>
          <div
            className="Import__title"
          >
            <h4>
              Add Project
            </h4>
          </div>
        </div>
        <div
          className="btn--import"
          onClick={[Function]}
        >
          Create New
        </div>
        <div
          className="hidden createLabbook"
        >
          <div
            className="ToolTip__toggle createLabbook"
            onClick={[Function]}
          >
            <div
              className="ToolTip__glow-container"
            >
              <div
                className="ToolTip__glow-ring-outer"
              >
                <div
                  className="ToolTip__glow-ring-inner"
                />
              </div>
            </div>
          </div>
          <div
            className="hidden createLabbook"
          />
          <div
            className="hidden createLabbook"
          >
            Clicking on the 'Create New' button will open the new Project window, letting you name and select a Base for the new Project.
          </div>
        </div>
        <div
          className="btn--import"
          onClick={[Function]}
        >
          Import Existing
        </div>
        <div
          className="hidden importLabbook"
        >
          <div
            className="ToolTip__toggle importLabbook"
            onClick={[Function]}
          >
            <div
              className="ToolTip__glow-container"
            >
              <div
                className="ToolTip__glow-ring-outer"
              >
                <div
                  className="ToolTip__glow-ring-inner"
                />
              </div>
            </div>
          </div>
          <div
            className="hidden importLabbook"
          />
          <div
            className="hidden importLabbook"
          >
            Clicking on the Import Existing' button transition the card to Import mode. 
             You can import by using a URL of a published Project or by uploading an exported Project's .zip file. 
             You can also directly import Projects that have been shared with you in the Cloud view.
          </div>
        </div>
      </div>
      <div
        className="hidden"
      />
    </div>
    <a
      className="Card Card--text column-4-span-3 flex flex--column justify--space-between"
      href="/projects/cbutler/zab"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__row--icons"
      >
        <div
          className="LocalLabbooks__containerStatus"
        >
          <button
            className="ContainerStatus__container-state LocalLabbooks__containerStatus--state loading box-shadow"
            onClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            
          </button>
        </div>
      </div>
      <div
        className="LocalLabbooks__row--text"
      >
        <div>
          <h6
            className="LocalLabbooks__panel-title"
            onClick={[Function]}
          >
            <span
              className={undefined}
            >
              <span
                className=""
                style={undefined}
              >
                zab
              </span>
            </span>
          </h6>
        </div>
        <p
          className="LocalLabbooks__paragraph LocalLabbooks__paragraph--owner "
        >
          Created by cbutler
        </p>
        <p
          className="LocalLabbooks__paragraph LocalLabbooks__paragraph--description"
        >
          <span
            className={undefined}
          />
        </p>
      </div>
      <div
        className="Tooltip LocalLabbookPanel__loading"
        data-tooltip="loading"
      />
    </a>
    <a
      className="Card Card--text column-4-span-3 flex flex--column justify--space-between"
      href="/projects/cbutler/untracked-labbook"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__row--icons"
      >
        <div
          className="LocalLabbooks__containerStatus"
        >
          <button
            className="ContainerStatus__container-state LocalLabbooks__containerStatus--state loading box-shadow"
            onClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            
          </button>
        </div>
      </div>
      <div
        className="LocalLabbooks__row--text"
      >
        <div>
          <h6
            className="LocalLabbooks__panel-title"
            onClick={[Function]}
          >
            <span
              className={undefined}
            >
              <span
                className=""
                style={undefined}
              >
                untracked-labbook
              </span>
            </span>
          </h6>
        </div>
        <p
          className="LocalLabbooks__paragraph LocalLabbooks__paragraph--owner "
        >
          Created by cbutler
        </p>
        <p
          className="LocalLabbooks__paragraph LocalLabbooks__paragraph--description"
        >
          <span
            className={undefined}
          />
        </p>
      </div>
      <div
        className="Tooltip LocalLabbookPanel__loading"
        data-tooltip="loading"
      />
    </a>
    <a
      className="Card Card--text column-4-span-3 flex flex--column justify--space-between"
      href="/projects/cbutler/untracked-2"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__row--icons"
      >
        <div
          className="LocalLabbooks__containerStatus"
        >
          <button
            className="ContainerStatus__container-state LocalLabbooks__containerStatus--state loading box-shadow"
            onClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            
          </button>
        </div>
      </div>
      <div
        className="LocalLabbooks__row--text"
      >
        <div>
          <h6
            className="LocalLabbooks__panel-title"
            onClick={[Function]}
          >
            <span
              className={undefined}
            >
              <span
                className=""
                style={undefined}
              >
                untracked-2
              </span>
            </span>
          </h6>
        </div>
        <p
          className="LocalLabbooks__paragraph LocalLabbooks__paragraph--owner "
        >
          Created by cbutler
        </p>
        <p
          className="LocalLabbooks__paragraph LocalLabbooks__paragraph--description"
        >
          <span
            className={undefined}
          />
        </p>
      </div>
      <div
        className="Tooltip LocalLabbookPanel__loading"
        data-tooltip="loading"
      />
    </a>
    <a
      className="Card Card--text column-4-span-3 flex flex--column justify--space-between"
      href="/projects/cbutler/untracked"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__row--icons"
      >
        <div
          className="LocalLabbooks__containerStatus"
        >
          <button
            className="ContainerStatus__container-state LocalLabbooks__containerStatus--state loading box-shadow"
            onClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            
          </button>
        </div>
      </div>
      <div
        className="LocalLabbooks__row--text"
      >
        <div>
          <h6
            className="LocalLabbooks__panel-title"
            onClick={[Function]}
          >
            <span
              className={undefined}
            >
              <span
                className=""
                style={undefined}
              >
                untracked
              </span>
            </span>
          </h6>
        </div>
        <p
          className="LocalLabbooks__paragraph LocalLabbooks__paragraph--owner "
        >
          Created by cbutler
        </p>
        <p
          className="LocalLabbooks__paragraph LocalLabbooks__paragraph--description"
        >
          <span
            className={undefined}
          />
        </p>
      </div>
      <div
        className="Tooltip LocalLabbookPanel__loading"
        data-tooltip="loading"
      />
    </a>
    <a
      className="Card Card--text column-4-span-3 flex flex--column justify--space-between"
      href="/projects/dmk/trump-tweets"
      onClick={[Function]}
    >
      <div
        className="LocalLabbooks__row--icons"
      >
        <div
          className="LocalLabbooks__containerStatus"
        >
          <button
            className="ContainerStatus__container-state LocalLabbooks__containerStatus--state loading box-shadow"
            onClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            
          </button>
        </div>
      </div>
      <div
        className="LocalLabbooks__row--text"
      >
        <div>
          <h6
            className="LocalLabbooks__panel-title"
            onClick={[Function]}
          >
            <span
              className={undefined}
            >
              <span
                className=""
                style={undefined}
              >
                trump-tweets
              </span>
            </span>
          </h6>
        </div>
        <p
          className="LocalLabbooks__paragraph LocalLabbooks__paragraph--owner "
        >
          Created by dmk
        </p>
        <p
          className="LocalLabbooks__paragraph LocalLabbooks__paragraph--description"
        >
          <span
            className={undefined}
          >
            <span
              className=""
              style={undefined}
            >
              A LabBook to analyze Trumps stupid tweets
            </span>
          </span>
        </p>
      </div>
      <div
        className="Tooltip LocalLabbookPanel__loading"
        data-tooltip="loading"
      />
    </a>
    <div
      className="LabbookPaginationLoader--hidden"
    >
      <div />
    </div>
    <div
      className="LabbookPaginationLoader--hidden"
    >
      <div />
    </div>
    <div
      className="LabbookPaginationLoader--hidden"
    >
      <div />
    </div>
    <div
      className="LabbookPaginationLoader--hidden"
    >
      <div />
    </div>
    <div
      className="LabbookPaginationLoader--hidden"
    >
      <div />
    </div>
  </div>
</div>
`;
